### PR TITLE
add refresh_expires_in property to credentials

### PR DESCRIPTION
**Pull request by Cantasura**
- Added the property `refresh_expires_in` property. This can help e.g. when the access token and refresh token is expired. With this property it is easier to handle the expiration of the refresh token. 
When the access token is expired and the lib tries to get a new access token with an expired refresh token, this will probably lead to and error.

---

- [✓] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by Cantasura** [comment]
can somebody please review this? :)

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@
 build/
 packages
 .packages
+.idea/
```
Comment (kevmoo) [inline .gitignore]: Please put this in your GLOBAL git ignore!

Comment (Cantasura) [reply .gitignore]: done :)

**Comment by kevmoo** [comment]
@devoncarew ?

**Comment by mosuem** [comment]
As this repo has moved to https://github.com/dart-lang/tools/tree/main/pkgs/oauth2, please refile the PR there. Thanks!

**Pull request closed without merging**